### PR TITLE
feat: remove type from auto-alarm

### PR DESCRIPTION
**Pull request by ufkaya**
BREAKING CHANGE

Backend requested to remove the type: "auto-alarm"
DH update API: https://github.com/carforyou/carforyou-dealerhub-web/pull/718

### CODE REVIEW

File: src/lib/toQueryString.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ const toQueryString = (params: object): string => {
   return Object.keys(params)
     .filter(
       (key) =>
-        typeof params[key] !== undefined &&
+        typeof params[key] !== "undefined" &&
```
Comment (masone) [inline src/lib/toQueryString.ts:6]: is this change related to removing the auto alarm type? judging from the commit history, you plan to squash-merge this, meaning it will end up in a `feat` commit

in projects, this is rarely an issue, but in packages, I'd be very explicit about the commit history. these will result in changelog entries after all. from reading the changelog, this will be an unexpected side effect

File: src/services/search/autoAlarm.ts [hunk 2]
```diff
@@ -6,25 +6,26 @@ import {
...
+    const query = toQueryString({ page, size })
     try {
       const result = await fetchPath(
         Service.SEARCH,
-        `dealers/${dealerId}/listing-saved-searches/auto-alarms?type=${type}${
-          page ? `&page=${page}` : ""
-        }${size ? `&page=${size}` : ""}`
+        `dealers/${dealerId}/listing-saved-searches/auto-alarms${
+          query ? `?${query}` : ""
+        }`
```
Comment (masone) [inline src/services/search/autoAlarm.ts:28]: same here :)

**Review by masone — CHANGES_REQUESTED** [review]
thanks for the valuable changes. if you clean up the commit history and rebase-merge, I'm fine with approving :) 

right now, there are two different changes in a commit history that you can't rebase, so I'm assuming you plan to squash

File: src/lib/toQueryString.ts (hunk 1 above)
Comment (ufkaya) [reply src/lib/toQueryString.ts:6]: this is a bug that I need to fix it so I can use this function, I can add it in the feat:

File: src/services/search/autoAlarm.ts (hunk 2 above)
Comment (ufkaya) [reply src/services/search/autoAlarm.ts:28]: What would be the right thing to do here? Do I need to create a PR to remove the type and another 1 to fix the bug?

File: src/lib/toQueryString.ts (hunk 1 above)
Comment (masone) [reply src/lib/toQueryString.ts:6]: allrighty, if it's directly correlated to your code change, let's roll with squashing it

Comment (masone) [reply src/lib/toQueryString.ts:6]: but it will change the behavior for other code that consumes this function and this will not be reflected in the changelog now. this decision is ultimately up to you

File: src/services/search/autoAlarm.ts (hunk 2 above)
Comment (masone) [reply src/services/search/autoAlarm.ts:28]: if they're independent of each other, that's what I would do. if you can only roll out the type when fixing the bug, I'd rewrite git history to contain two commits - one to `fix` the bug, one for the `feat` and then instead of squashing, do a rebase merge

**Review by masone — APPROVED** [review]
unblocking the PR now after we had the discussion on commit granularity / git history / changelog :) 
how you want to execute is up to you. there are multiple valid ways to go about it
